client/tests/test_training_set_dataframe.py

```diff
@@ -0,0 +1,143 @@
+import time
+import pytest
+import pandas as pd
+import featureform as ff
+from featureform import Client
+
+
+@pytest.mark.parametrize(
+    "provider_source_fxt,is_local,is_insecure",
+    [
+        pytest.param(
+            "local_provider_source",
+            True,
+            True,
+            marks=pytest.mark.local,
+        ),
+        pytest.param(
+            "hosted_sql_provider_and_source",
+            False,
+            False,
+            marks=pytest.mark.hosted,
+        ),
+        pytest.param(
+            "hosted_sql_provider_and_source",
+            False,
+            True,
+            marks=pytest.mark.docker,
+        ),
+    ],
+)
+def test_training_set_as_dataframe(provider_source_fxt, is_local, is_insecure, request):
+    custom_marks = [
+        mark.name for mark in request.node.own_markers if mark.name != "parametrize"
+    ]
+    provider, source, inference_store = request.getfixturevalue(provider_source_fxt)(
+        custom_marks
+    )
+
+    # Arranges the resources context following the Quickstart pattern
+    ts_name, ts_variant = arrange_resources(
+        provider, source, inference_store, is_local, is_insecure
+    )
+
+    client = Client(local=is_local, insecure=is_insecure)
+    ts_df = client.training_set(ts_name, ts_variant).dataframe()
+
+    assert isinstance(ts_df, pd.DataFrame) and len(ts_df) > 0
```

Comment:
do we need to assert expected values here? currently it's only checking the type and length. skip if otherwise.